webapp/channels/src/components/advanced_text_editor/send_button/send_post_options/index.tsx

```diff
@@ -0,0 +1,175 @@
+import Timestamp from 'components/timestamp';
+
+import './style.scss';
+
+type Props = {
+    disabled?: boolean;
+    onSelect: (e: React.FormEvent, schedulingInfo: SchedulingInfo) => void;
+}
+
+export function SendPostOptions({disabled, onSelect}: Props) {
+    const {formatMessage} = useIntl();
+
+    const handleOnSelect = useCallback((e: React.FormEvent, scheduledAt: number) => {
+        const schedulingInfo: SchedulingInfo = {
+            scheduled_at: scheduledAt,
+        };
+
+        onSelect(e, schedulingInfo);
+    }, [onSelect]);
+
+    const coreMenuOptions = useMemo(() => {
+        const today = new Date();
+        today.setHours(9, 0, 0, 0); // 9 AM
+        const unixTimestamp = today.getTime();
+
+        const timeComponent = (
+            <Timestamp
+                value={unixTimestamp}
+                useDate={false}
+            />
+        );
+
+        const tomorrow9amTime = new Date();
+        tomorrow9amTime.setHours(9, 0, 0, 0);
+        tomorrow9amTime.setDate(today.getDate() + 1);
+
+        const optionTomorrow = (
+            <Menu.Item
+                onClick={(e) => handleOnSelect(e, tomorrow9amTime.getTime())}
+                labels={
+                    <FormattedMessage
+                        id='create_post_button.option.schedule_message.options.tomorrow'
+                        defaultMessage='Tomorrow at {9amTime}'
+                        values={{'9amTime': timeComponent}}
+                    />
+                }
+            />
+        );
+
+        const nextMonday = new Date();
+        const dayOfWeek = today.getDay();
+        const daysUntilNextMonday = (8 - dayOfWeek) % 7 || 7;
+        nextMonday.setDate(today.getDate() + daysUntilNextMonday);
+        nextMonday.setHours(9, 0, 0, 0);
+
+        const optionNextMonday = (
+            <Menu.Item
+                onClick={(e) => handleOnSelect(e, nextMonday.getTime())}
+                labels={
+                    <FormattedMessage
+                        id='create_post_button.option.schedule_message.options.next_monday'
+                        defaultMessage='Next Monday at {9amTime}'
+                        values={{'9amTime': timeComponent}}
+                    />
+                }
+            />
+        );
+
+        const optionMonday = (
+            <Menu.Item
+                onClick={(e) => handleOnSelect(e, nextMonday.getTime())}
+                labels={
+                    <FormattedMessage
+                        id='create_post_button.option.schedule_message.options.monday'
+                        defaultMessage='Monday at {9amTime}'
+                        values={{
+                            '9amTime': timeComponent,
+                        }}
+                    />
+                }
+            />
+        );
+
+        const options: React.ReactElement[] = [];
+
+        switch (today.getDay()) {
+        // Sunday
+        case 0:
+            options.push(optionTomorrow);
```

Comment:
Nitpicking, but it might be nicer to right this as
```js
let options;

switch (today.getDay()) {
case 0:
    options = [optionTomorrow];
    break;
    ...
}

return options;
```

I think that should still type correctly, and I think using the array literals is a bit nicer than pushing onto the array.

Reply:
Done.